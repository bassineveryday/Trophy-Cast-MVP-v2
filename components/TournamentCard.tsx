import React from 'react';
import { Platform, Pressable, StyleSheet, Text, View } from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';
import { Ionicons } from '@expo/vector-icons';
import { useTheme } from '../lib/ThemeContext';
import type { BrandTheme } from '../lib/ThemeContext';

export type TournamentCardProps = {
  id?: string | number;
  title: string;
  lake?: string;
  date?: string;
  location?: string;
  attending?: number;
  variant?: 'gold' | 'green' | 'teal' | 'blue';
  onPress?: (id?: string | number) => void;
  testID?: string;
};

const getVariantGradient = (theme: BrandTheme, variant: string): [string, string] => {
  switch (variant) {
    case 'gold':
      return theme.gradients.accent;
    case 'green':
      return [theme.success, theme.primary];
    case 'teal':
      return theme.gradients.hero;
    case 'blue':
    default:
      return [theme.primaryLight, theme.primary];
  }
};

export default function TournamentCard({
  id,
  title,
  lake,
  date,
  location,
  attending = 0,
  variant = 'teal',
  onPress,
  testID,
}: TournamentCardProps) {
  const { theme } = useTheme();
  const styles = React.useMemo(() => createStyles(theme), [theme]);
  const colors = getVariantGradient(theme, variant);

  const content = (
    <View style={styles.inner}>
      <View style={styles.left}>
        <View style={styles.thumbPlaceholder}>
          <Ionicons name="fish" size={28} color={theme.onPrimary} />
        </View>
      </View>

      <View style={styles.body}>
        <Text style={styles.title} numberOfLines={1}>{title}</Text>
        {!!lake && <Text style={styles.meta}>{lake}{location ? ` â€¢ ${location}` : ''}</Text>}
        {!!date && <Text style={styles.meta}>{date}</Text>}
      </View>

      <View style={styles.right}>
        <View style={styles.attendBadge}>
          <Text style={styles.attendText}>{attending}</Text>
        </View>
      </View>
    </View>
  );

  return Platform.OS === 'web' ? (
    <Pressable
      onPress={() => onPress?.(id)}
      testID={testID}
      accessibilityRole="button"
      accessibilityLabel={`${title} tournament card`}
      accessibilityHint="Opens tournament details"
      style={[styles.container, { backgroundColor: theme.primary }]}
    >
      {content}
    </Pressable>
  ) : (
    <LinearGradient colors={colors} start={[0,0]} end={[1,1]} style={styles.nativeGradient}>
      <Pressable
        onPress={() => onPress?.(id)}
        testID={testID}
        accessibilityRole="button"
        accessibilityLabel={`${title} tournament card`}
        accessibilityHint="Opens tournament details"
        style={styles.pressable}
      >
        {content}
      </Pressable>
    </LinearGradient>
  );
}

function withAlpha(hexOrRgb: string, alpha: number) {
  if (hexOrRgb.startsWith('#')) {
    const hex = hexOrRgb.replace('#', '');
    const full = hex.length === 3 ? hex.split('').map(c => c + c).join('') : hex;
    const bigint = parseInt(full, 16);
    const r = (bigint >> 16) & 255;
    const g = (bigint >> 8) & 255;
    const b = bigint & 255;
    return `rgba(${r}, ${g}, ${b}, ${alpha})`;
  }
  return hexOrRgb;
}

function createStyles(theme: BrandTheme) {
  return StyleSheet.create({
    container: {
      borderRadius: 12,
      marginBottom: 12,
      overflow: 'hidden',
      padding: 12,
    },
    nativeGradient: {
      borderRadius: 12,
      marginBottom: 12,
      overflow: 'hidden',
    },
    pressable: {
      padding: 12,
    },
    inner: {
      flexDirection: 'row',
      alignItems: 'center',
    },
    left: {
      marginRight: 12,
    },
    thumbPlaceholder: {
      width: 56,
      height: 56,
      borderRadius: 8,
      backgroundColor: withAlpha(theme.onPrimary, 0.08),
      alignItems: 'center',
      justifyContent: 'center',
    },
    body: {
      flex: 1,
    },
    title: {
      color: theme.onPrimary,
      fontSize: 16,
      fontWeight: '700',
      marginBottom: 4,
    },
    meta: {
      color: withAlpha(theme.onPrimary, 0.85),
      fontSize: 12,
    },
    right: {
      marginLeft: 8,
    },
    attendBadge: {
      backgroundColor: withAlpha(theme.onPrimary, 0.12),
      paddingHorizontal: 10,
      paddingVertical: 6,
      borderRadius: 20,
    },
    attendText: {
      color: theme.onPrimary,
      fontWeight: '700',
    },
  });
}
