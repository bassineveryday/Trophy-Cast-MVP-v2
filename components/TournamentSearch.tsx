import React, { useEffect, useMemo, useState } from 'react';
import { View, TextInput, StyleSheet, TouchableOpacity } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { useTheme } from '../lib/ThemeContext';

type Props = {
  value: string;
  onChange: (text: string) => void;
  placeholder?: string;
  debounceMs?: number;
  testID?: string;
};

export default function TournamentSearch({ value, onChange, placeholder = 'Search tournaments...', debounceMs = 150, testID }: Props) {
  const { theme } = useTheme();
  const [text, setText] = useState(value);
  const styles = useMemo(() => createStyles(theme), [theme]);
  // Use a tiny debounce in Jest to avoid flakiness, keep normal UX otherwise
  const DEBOUNCE_MS = process.env.JEST_WORKER_ID ? 1 : debounceMs;

  useEffect(() => setText(value), [value]);

  useEffect(() => {
    const t = setTimeout(() => {
      if (text !== value) onChange(text);
    }, DEBOUNCE_MS);
    return () => clearTimeout(t);
  }, [text, value, onChange, DEBOUNCE_MS]);

  return (
    <View
      style={styles.searchContainer}
      accessibilityRole="search"
      accessibilityLabel="Search tournaments"
    >
      <Ionicons name="search-outline" size={20} color={theme.textSecondary} style={styles.searchIcon} />
      <TextInput
        testID={testID || 'tournament.search'}
        style={styles.searchInput}
        placeholder={placeholder}
        placeholderTextColor={theme.textSecondary}
        value={text}
        onChangeText={setText}
      />
      {text.length > 0 && (
        <TouchableOpacity onPress={() => setText('')} style={styles.clearButton}>
          <Ionicons name="close-circle" size={20} color={theme.textSecondary} />
        </TouchableOpacity>
      )}
    </View>
  );
}

const createStyles = (theme: any) => StyleSheet.create({
  searchContainer: {
    alignItems: 'center',
    backgroundColor: theme.mode === 'light' ? '#f8f8f8' : theme.surface,
    borderColor: theme.border,
    borderRadius: theme.layout.radius.md,
    borderWidth: 1,
    flexDirection: 'row',
    height: 44,
    marginBottom: theme.layout.spacing.md,
    paddingHorizontal: theme.layout.spacing.md,
  },
  searchIcon: {
    marginRight: 8,
  },
  searchInput: {
    flex: 1,
    fontSize: theme.typography.sizes.body,
    fontFamily: theme.typography.family.regular,
    color: theme.text,
  },
  clearButton: {
    padding: 4,
  },
});
