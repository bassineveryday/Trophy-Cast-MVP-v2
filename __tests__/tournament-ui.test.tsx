import React from 'react';
import { render, fireEvent, act } from '@testing-library/react-native';
import TournamentCard from '../components/TournamentCard';
import TournamentSearch from '../components/TournamentSearch';

describe('Tournament UI components', () => {
  it('TournamentCard renders basic info and is pressable', () => {
    const onPress = jest.fn();
    const { toJSON, getByTestId } = render(
      <TournamentCard
        id="evt-1"
        title="Norton Lake Bash"
        lake="Norton"
        date="Sat, Sep 13, 2025"
        attending={24}
        onPress={onPress}
        testID="card.norton"
      />
    );
    expect(toJSON()).toMatchSnapshot();
    fireEvent.press(getByTestId('card.norton'));
    expect(onPress).toHaveBeenCalledWith('evt-1');
  });

  it('TournamentSearch debounces calls to onChange', () => {
    jest.useFakeTimers();
    const onChange = jest.fn();
    const { getByTestId } = render(
      <TournamentSearch value="" onChange={onChange} debounceMs={200} />
    );
    const input = getByTestId('tournament.search');

    act(() => {
      fireEvent.changeText(input, 'Nor');
      fireEvent.changeText(input, 'Nort');
      fireEvent.changeText(input, 'Norton');
      jest.advanceTimersByTime(199);
    });
    expect(onChange).not.toHaveBeenCalled();

    act(() => {
      jest.advanceTimersByTime(1);
    });
    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange).toHaveBeenCalledWith('Norton');

    jest.useRealTimers();
  });
});
