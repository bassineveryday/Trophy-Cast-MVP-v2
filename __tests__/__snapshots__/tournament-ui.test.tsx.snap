// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tournament UI components TournamentCard renders basic info and is pressable 1`] = `
<View
  style={
    {
      "borderRadius": 12,
      "marginBottom": 12,
      "overflow": "hidden",
    }
  }
>
  <View
    accessibilityHint="Opens tournament details"
    accessibilityLabel="Norton Lake Bash tournament card"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "padding": 12,
      }
    }
    testID="card.norton"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          {
            "marginRight": 12,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "rgba(255, 255, 255, 0.08)",
              "borderRadius": 8,
              "height": 56,
              "justifyContent": "center",
              "width": 56,
            }
          }
        >
          <Text
            data-testid="icon-fish"
          >
            fish
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "flex": 1,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            {
              "color": "#ffffff",
              "fontSize": 16,
              "fontWeight": "700",
              "marginBottom": 4,
            }
          }
        >
          Norton Lake Bash
        </Text>
        <Text
          style={
            {
              "color": "rgba(255, 255, 255, 0.85)",
              "fontSize": 12,
            }
          }
        >
          Norton
        </Text>
        <Text
          style={
            {
              "color": "rgba(255, 255, 255, 0.85)",
              "fontSize": 12,
            }
          }
        >
          Sat, Sep 13, 2025
        </Text>
      </View>
      <View
        style={
          {
            "marginLeft": 8,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "rgba(255, 255, 255, 0.12)",
              "borderRadius": 20,
              "paddingHorizontal": 10,
              "paddingVertical": 6,
            }
          }
        >
          <Text
            style={
              {
                "color": "#ffffff",
                "fontWeight": "700",
              }
            }
          >
            24
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
